plugins/module_utils/iam.py

```diff
@@ -27,6 +28,68 @@ class AnsibleIAMError(AnsibleAWSError):
     pass
 
 
+def iam_common_error_handler(description):
+    """A simple error handler that catches the standard Boto3 exceptions and raises
```

Comment:
I was only wondering if we could make it more generic and not only for IAM. I like the decorator's approach for error handling.

Reply:
I was wondering the same thing, I didn't want to dig too deep if they wasn't a direction the wider team wanted things to go, but can certainly poke at it.